src/main/scala/org/broadinstitute/dsde/workbench/avram/api/AvramServlet.scala

```diff
@@ -14,60 +16,58 @@ import org.broadinstitute.dsde.workbench.avram.util.AvramResult.unsafeRun
 import scala.concurrent.ExecutionContext
 
 
-trait AvramServlet {
+abstract class AvramServlet(avram: Avram) {
 
   implicit val executionContext: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
 
   val log: Logger = Logger.getLogger(getClass.getName)
-  private val samDao = Avram.samDao
+  private val samDao = avram.samDao
   private val bearerPattern = """(?i)bearer (.*)""".r
 
-  def handleAuthenticatedRequest[T](request: HttpServletRequest, response: HttpServletResponse)
+  def handleAuthenticatedRequest[T](bearerToken: String)
                                    (f: SamUserInfoResponse => AvramResult[T])
-                                   (implicit encoder: Encoder[T]): Unit = {
-    def writeBody(body: T): Unit = {
-      response.setStatus(HttpServletResponse.SC_OK)
-      response.setContentType("application/json")
-      // CORS?
-      response.getWriter.write(body.asJson.noSpaces)
+                                   (implicit encoder: Encoder[T]): Response = {
+    def writeBody(body: T): Response = {
+      Response
+        .status(Status.OK)
+        .entity(body.asJson.noSpaces)
```

Comment:
Did you lose the `application/json` content type header?

Reply:
Added @produces annotation